src/stravalib/model.py

```diff
@@ -87,22 +96,29 @@ def naive_datetime(value: Optional[AllDateTypes]) -> Optional[datetime]:
 
     YYYY-MM-DD[T]HH:MM[:SS[.ffffff]][Z or [±]HH[:]MM]
     """
+
     if not value:
         return None
 
-    # Allow for timestamps to be passed as strings, e.g '1714305600'.
-    try:
-        value_is_int = isinstance(value, int) or float(value).is_integer()
-    except (ValueError, TypeError):
-        value_is_int = False
-
-    dt = (
-        parser.parse(str(value))
-        if not value_is_int
-        else datetime.fromtimestamp(int(value))
-    )
-    # Remove timezone information, if any
-    return dt.replace(tzinfo=None)
```

Comment:
The `float()` and `int()` calls caused typing issues. We needed some more verbose `isinstance()` checks.